test/integration/deployment/deployment_test.go

```diff
@@ -969,26 +970,31 @@ func TestDeploymentAvailableCondition(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// Verify all replicas fields of DeploymentStatus have desired counts
+	if err = tester.checkDeploymentStatusReplicasFields(10, 10, 0, 0, 10); err != nil {
+		t.Fatal(err)
+	}
+
 	// Mark the pods as ready without waiting for the deployment to complete
-	pods, err := tester.listUpdatedPods()
-	if err != nil {
+	if err = tester.markUpdatedPodsReadyWithoutComplete(); err != nil {
 		t.Fatal(err)
 	}
-	for i := range pods {
-		pod := pods[i]
-		if podutil.IsPodReady(&pod) {
-			continue
-		}
-		if err = markPodReady(c, ns.Name, &pod); err != nil {
-			t.Fatalf("failed to update Deployment pod %q, will retry later: %v", pod.Name, err)
-		}
+
+	// Wait for number of ready replicas to equal number of replicas.
+	if err = tester.waitForReadyReplicas(); err != nil {
```

Comment:
This is necessary to prevent race when determing number of ready replicas in next step.